src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
+      const newComment = {
+        name: name.trim(),
+        email: email.trim(),
+        body: body.trim(),
+        postId: selectedPost?.id,
+      };
 
-    if (!name || !email || !body) {
-      return;
-    }
+      setErrors({
+        name: !newComment.name,
+        email: !newComment.email,
+        body: !newComment.body,
+      });
+
+      if (!newComment.name || !newComment.email || !newComment.body) {
+        return;
+      }
 
-    setSubmitting(true);
+      setSubmitting(true);
 
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+      await dispatch(addNewComment(newComment));
 
-    // and the spinner will disappear immediately
-    setSubmitting(false);
-    setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
+      setSubmitting(false);
+      setValues(current => ({ ...current, body: '' }));
```

Comment:
It looks like you're only clearing the `body` field after submitting the form. If this is the intended behavior, it would be helpful to include a comment explaining why. If not, you should clear all fields.